applications/StructuralMechanicsApplication/custom_processes/set_moving_load_process.cpp

```diff
@@ -0,0 +1,319 @@
+    }
+
+    // find non repeating node ids
+    std::vector<int> non_repeating_node_ids = FindNonRepeatingIndices(node_id_vector);
+
+    // error if model part does not have 1 end and 1 beginning
+    KRATOS_ERROR_IF_NOT(non_repeating_node_ids.size() ==2) << "Moving load condition model part needs to be connected with a beginning and end" << std::endl;
+    
+    
+
+    // find conditions at both ends of modelpart
+    std::vector<Condition> end_conditions;
+    for (Condition& r_cond : mrModelPart.Conditions()) {
+
+        auto& r_geom = r_cond.GetGeometry();
+
+        for (int i = 0; i < r_geom.size(); i++)
+        {
+            for (int j = 0; j < non_repeating_node_ids.size(); j++)
+            {
+                if (r_geom[i].Id() == non_repeating_node_ids[j])
+                {
+                    end_conditions.push_back(r_cond);
+                }
+            }    
+        }
+    }
+
+
+    // find start condition 
+    Point center_1 = end_conditions[0].GetGeometry().Center();
+    Point center_2 = end_conditions[1].GetGeometry().Center();
+
+    Condition& r_first_cond = GetFirstCondition(center_1, center_2, direction, end_conditions);
+
+    mSortedConditions = BubbleSortConditions(mrModelPart.Conditions(), r_first_cond);
+
+    auto tmp = r_first_cond.GetGeometry();
+
+    int a = 1 + 1;
+    //for (int i = 0; i < arr_size; i++) {
+    //    node_id_vector[node_id_vector[i] % arr_size]
+    //        = node_id_vector[node_id_vector[i] % arr_size] + arr_size;
+    //}
+
+
+    //IntervalUtility interval_utility(mParameters);
+   /* if (interval_utility.IsInInterval(current_time)) {
+        double total_area = 0.0;
+        for (auto& r_cond : mrModelPart.Conditions()) {
+            total_area += r_cond.GetGeometry().Area();
+        }
+
+        const double global_total_area = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(total_area);
+
+        Vector force_by_area = mParameters["load"].GetVector() / global_total_area;
+
+        for (auto& r_cond : mrModelPart.Conditions()) {
+            const double area = r_cond.GetGeometry().Area();
+            r_cond.SetValue(SURFACE_LOAD, force_by_area * area);
+        }
+    }*/
+    KRATOS_CATCH("")
+}
+
+void SetMovingLoadProcess::ExecuteInitializeSolutionStep()
+{
+    double distance_cond = 0;
+
+
+    // bool to check if load is already added, such that a load is not added twice if the load is exactly at a shared node.
+    bool is_moving_load_added = false;
+
+    for (auto& r_cond : mSortedConditions)
+    {
+        auto& r_geom = r_cond.GetGeometry();
+        double element_length = r_geom.Length();
+
+        // if moving load is located at current condition element, apply moving load, else apply a zero load
+        if ((distance_cond + element_length >= mCurrentDistance) && (distance_cond <= mCurrentDistance) && !is_moving_load_added)
+        {
+            
+            r_cond.SetValue(MOVING_LOAD, mLoad);
```

Comment:
I don't think it is a good idea to create a new `MOVING_LOAD` variable as you'll require to update all the conditions in order to assemble it to the local system. Current Kratos load variables should be used instead.

Reply:
OK. Now I see, you created your custom condition.

Reply:
In any case, I'd still use `POINT_LOAD`.

Reply:
I will look into that